hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -74,14 +74,15 @@
 
   int immutCellsNum = 0;  // number of immutable for compaction cells
   /**
-   * Types of Compaction
+   * Types of actions to be done on the pipeline upon MemStoreCompaction invocation
    */
-  private enum Type {
-    COMPACT_TO_SKIPLIST_MAP,
-    COMPACT_TO_ARRAY_MAP
+  private enum Action {
```

Comment:
perhaps you already changed this ... , but I think 
at this point we can discard COMPACT_TO_SKIPLIST_MAP -- we know array map is working well;
so we are left with the distinction between merging vs. compaction, which can be named MERGE_INDICES, COMPACT_DATA
Shouldn't we have here also the option for flattening? call it FLATTEN_INDEX
and the distinction between array map and chunk map will be derived from the segment itself, no?